java/code/src/com/suse/manager/webui/controllers/shared/channels/ChannelsUtils.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2019 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.manager.webui.controllers.shared.channels;
```

Comment:
Hmm, we already have a `com.suse.manager.webui.controllers.utils`, wouldn't it make sense to move this class there?

Reply:
Hum I really like to split things by feature/area rather than by function. For me it makes more sense to have something like:

channels
    - utils.java
    - logic.java
another_feature
    - utils.java
    - logic.java


instead of:

utils
    channels.utils.java
    another_feature.utils.java
    another_feature2.utils.java


what do you think ?

Reply:
Yeah, those are 2 views to look at that. Yours is totally valid (although it'd lead to explosion of number of packages - mostly we would have a package for each feature that would only contain single controller...). I'd prefer sticking to the convention as we already have:

`webui.controllers`:
- `ImageController`
- `AKController`
- `webui.controllers.utils`:
  - `ImageUtils`
  - `AKUtils`

I wouldn't like mixing those 2 approaches - we should stick to one. Feel free to propose refactoring the `webui.controllers` package on the mailing list. Anyways - this should be solved separately from this bugfix.

Reply:
I moved everything to "controllers/channels" does that look better?

It makes sense to have everything together that corresponds to one feature :)

The controllers folder is growing uncontrolled and it's getting hard to understand :D